src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -745,18 +797,42 @@ await ExecuteWithFailOverPolicyAsync(
 
                 Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
                 SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+
+                List<KeyValueWatcher> selectedKeyValueWatchers = new List<KeyValueWatcher>();
+
+                if (_options.RefreshOptions.RegisterAllEnabled)
+                {
+                    foreach (KeyValueIdentifier keyValueIdentifier in watchedSelectedKeyValueCollections.Keys)
+                    {
+                        selectedKeyValueWatchers.AppendUnique(new KeyValueWatcher()
+                        {
+                            Key = keyValueIdentifier.Key,
+                            Label = keyValueIdentifier.Label,
+                            RefreshInterval = _options.RefreshOptions.RefreshInterval
+                        });
+                    }
+                }
+
+                // Update the next refresh time for all refresh registered settings and feature flags
+                foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers.Concat(_options.FeatureFlagWatchers).Concat(selectedKeyValueWatchers))
+                {
+                    UpdateNextRefreshTime(changeWatcher);
```

Comment:
We should be calling `UpdateNextRefreshTime` even if `data` is null

Reply:
I'm trying to see where this ends up being an issue. I know we call `InitializaAsync` again in `RefreshAsync` if there's no data, but is there a case where `data` is null and we still rely on `NextRefreshTime`? It seems like we check to see if there are any refreshable watchers, but that information isn't used if `_mappedData` is null since we just try to initialize everything again.

Reply:
Moved back to original spot